components/aws/sagemaker/ground_truth/src/sagemaker_ground_truth_component.py

```diff
@@ -0,0 +1,348 @@
+                )
+        else:
+            logging.error(
+                "Task type must be Bounding Box, Image Classification, Semantic Segmentation, Text Classification, or Custom."
+            )
+
+        request["HumanTaskConfig"]["UiConfig"]["UiTemplateS3Uri"] = inputs.ui_template
+        request["HumanTaskConfig"]["TaskTitle"] = inputs.title
+        request["HumanTaskConfig"]["TaskDescription"] = inputs.description
+        request["HumanTaskConfig"][
+            "NumberOfHumanWorkersPerDataObject"
+        ] = inputs.num_workers_per_object
+        request["HumanTaskConfig"]["TaskTimeLimitInSeconds"] = inputs.time_limit
+
+        if inputs.task_availibility:
+            request["HumanTaskConfig"][
+                "TaskAvailabilityLifetimeInSeconds"
+            ] = inputs.task_availibility
+        else:
+            request["HumanTaskConfig"].pop("TaskAvailabilityLifetimeInSeconds")
+
+        if inputs.max_concurrent_tasks:
+            request["HumanTaskConfig"][
+                "MaxConcurrentTaskCount"
+            ] = inputs.max_concurrent_tasks
+        else:
+            request["HumanTaskConfig"].pop("MaxConcurrentTaskCount")
+
+        if inputs.task_keywords:
+            for word in [n.strip() for n in inputs.task_keywords.split(",")]:
+                request["HumanTaskConfig"]["TaskKeywords"].append(word)
+        else:
+            request["HumanTaskConfig"].pop("TaskKeywords")
+
+        ### Update worker configurations
+        if inputs.worker_type.lower() == "public":
+            if inputs.no_adult_content:
+                request["InputConfig"]["DataAttributes"]["ContentClassifiers"].append(
+                    "FreeOfAdultContent"
+                )
+            if inputs.no_ppi:
+                request["InputConfig"]["DataAttributes"]["ContentClassifiers"].append(
+                    "FreeOfPersonallyIdentifiableInformation"
+                )
+
+            request["HumanTaskConfig"][
+                "WorkteamArn"
+            ] = "arn:aws:sagemaker:{}:394669845002:workteam/public-crowd/default".format(
+                inputs.region
+            )
+
+            dollars = int(inputs.workforce_task_price)
+            cents = int(100 * (inputs.workforce_task_price - dollars))
+            tenth_of_cents = int(
+                (inputs.workforce_task_price * 1000) - (dollars * 1000) - (cents * 10)
+            )
+            request["HumanTaskConfig"]["PublicWorkforceTaskPrice"]["AmountInUsd"][
+                "Dollars"
+            ] = dollars
+            request["HumanTaskConfig"]["PublicWorkforceTaskPrice"]["AmountInUsd"][
+                "Cents"
+            ] = cents
+            request["HumanTaskConfig"]["PublicWorkforceTaskPrice"]["AmountInUsd"][
+                "TenthFractionsOfACent"
+            ] = tenth_of_cents
+        else:
+            request["InputConfig"].pop("DataAttributes")
+            request["HumanTaskConfig"]["WorkteamArn"] = inputs.workteam_arn
```

Comment:
For later: Replace hardcoded request parameters for all components in a generic way